clang/lib/Sema/SemaOverload.cpp

```diff
@@ -10968,6 +10976,16 @@ bool clang::isBetterOverloadCandidate(
            S.CUDA().IdentifyPreference(Caller, Cand2.Function);
   }
 
+  // In SYCL device compilation mode prefer the overload with the
+  // SYCLDeviceOnly attribute.
+  if (S.getLangOpts().isSYCL() && S.getLangOpts().SYCLIsDevice &&
```

Comment:
Is `SYCLIsDevice` ever true while `isSYCL` is false? I wonder if this could be simplified to just `S.getLangOpts().SYCLIsDevice`.